src/index.js

```diff
@@ -298,12 +298,12 @@ class DropdownTreeSelect extends Component {
             { 'radio-select': mode === 'radioSelect' }
           )}
         >
-          <Trigger onTrigger={this.onTrigger} showDropdown={showDropdown} {...commonProps}>
+          <Trigger onTrigger={this.onTrigger} showDropdown={showDropdown} {...commonProps} tags={tags}>
```

Comment:
` tags={tags}` Are using tags for populating aria attributes? If so, then:

 - why does it need to know since the trigger is not rendering the tags
 - how about pulling up the `tags > label logic` & pass the labels?

Reply:
Well the tags are rendered/labeled/id:ed in the child node/Input. So it would be if I would ref them or bundle them (Input and Trigger) more "together". Instead of passing the input as a child one could render the Input within the trigger-component instead? Or what are you suggesting?

Reply:
The trigger is a separate component, it doesn't need to know anything about input. The reason there is a parent-child relationship right now is for backward compatibility. I was using a 3rd party component which outputted the current HTML structure. When I replaced that 3rd party with in-house, I kept the same structure. 

The main component (index.js) is the owner and currently houses most business logic. Individual components only concern themselves with presentational logic. That was roughly the idea/organization that I started with. 

We could just break free of legacy baggage and redesign this in a manner where Trigger doesn't appear to be Input's parent. 

If we do not want to cause a major change, then we could calculate aria attributes for tags and pass them to tags as well as Trigger component. Thing is if we go this route, we're breaking the encapsulation of presentational logic of a component within the component (aria attributes of tags being calculated outside). That'd be acceptable if we do it for all components - so aria attributes are calculated outside of each component and then passed on to them as props. 

I'm not sure which one is better, but now you know a bit of context. Like always, all suggestions welcome! :)

Reply:
Not sure how I could do this simpler that it is with its current structure and did not have the intention of rewriting the component, just fix so that the correct references are set between the "focus node" and its selected tags. And those are generated two steps down (the ${_id}_tag of the tags based on passed state.tags). 

The only idea would be to have a function which "bubbles" up the generated tag ids or references to pass to the parent. But not sure how much of a difference that would make...

I made a shared function for tag id generation so that's done in a single way at least.

Reply:
Hear ya… Let's go with this now and revisit this later.